engine/src/test/java/io/camunda/zeebe/engine/processing/variable/mapping/NoneEndEventOutputMappingTest.java

```diff
@@ -0,0 +1,222 @@
+        scopeVariables(variable("y", "true"))
+      },
+      {
+        "{'x': %s}".formatted(A_STRING),
+        mapping(b -> b.zeebeOutputExpression("substring(x, 4)", "y")),
+        scopeVariables(variable("y", A_SUB_STRING))
+      },
+      {
+        "{'x': %s}".formatted(A_STRING),
+        mapping(b -> b.zeebeOutputExpression("upper case(x)", "y")),
+        scopeVariables(variable("y", A_UPPER_STRING))
+      },
+      {
+        "{'x': %s}".formatted(A_TIME_VALUE),
+        mapping(b -> b.zeebeOutputExpression("time(x)", "y")),
+        scopeVariables(variable("y", A_TIME_VALUE))
+      },
+      {
+        "{'x': %s}".formatted(A_LOCAL_TIME_VALUE),
+        mapping(b -> b.zeebeOutputExpression("time(x)", "y")),
+        scopeVariables(variable("y", A_LOCAL_TIME_VALUE))
+      },
+      {
+        "{'x': %s}".formatted(A_DATE_VALUE),
+        mapping(b -> b.zeebeOutputExpression("date(x)", "y")),
+        scopeVariables(variable("y", A_DATE_VALUE))
+      },
+      {
+        "{'x': %s}".formatted(A_LOCAL_DATE_AND_TIME_VALUE),
+        mapping(b -> b.zeebeOutputExpression("date and time(x)", "y")),
+        scopeVariables(variable("y", A_LOCAL_DATE_AND_TIME_VALUE))
+      },
+      {
+        "{'x': %s}".formatted(A_DATE_AND_TIME_VALUE),
+        mapping(b -> b.zeebeOutputExpression("date and time(x)", "y")),
+        scopeVariables(variable("y", A_DATE_AND_TIME_VALUE))
+      },
+      {
+        "{'x': %s}".formatted(A_DAY_TIME_DURATION_VALUE),
+        mapping(b -> b.zeebeOutputExpression("duration(x)", "y")),
+        scopeVariables(variable("y", A_DAY_TIME_DURATION_VALUE))
+      },
+      {
+        "{'x': %s}".formatted(A_YEAR_MONTH_DURATION_VALUE),
+        mapping(b -> b.zeebeOutputExpression("duration(x)", "y")),
+        scopeVariables(variable("y", A_YEAR_MONTH_DURATION_VALUE))
+      }
+    };
+  }
+
+  @Test
+  public void shouldApplyOutputMappings() {
+    // given
+    ENGINE_RULE
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess(PROCESS_ID)
+                .startEvent()
+                .endEvent(
+                    "end",
+                    builder -> {
+                      mappings.accept(builder);
+                    })
+                .done())
+        .deploy()
+        .getValue()
+        .getProcessesMetadata()
+        .get(0)
+        .getProcessDefinitionKey();
+
+    // when
+    final long processInstanceKey =
+        ENGINE_RULE
+            .processInstance()
+            .ofBpmnProcessId(PROCESS_ID)
+            .withVariables(initialVariables)
+            .create();
+
+    // then
+    final long latestPayloadVariablePosition =
+        RecordingExporter.variableRecords(VariableIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .filter(r -> initialVariables.contains(r.getValue().getName()))
+            .map(Record::getPosition)
+            .max(Comparator.naturalOrder())
+            .orElseThrow();
```

Comment:
❌ You may have noticed executing these tests takes a long time. This is because we do not have a limit when searching for the `latestPayloadVariablePosition`. It's hard to define a proper limit for this, but we could make this simpler by getting the position of the end event `ACTIVATED` event instead.

```
    final long endEventActivatePosition =
        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
            .withProcessInstanceKey(processInstanceKey)
            .withElementId("end")
            .limit(1)
            .getFirst()
            .getPosition();
```